lms/djangoapps/grades/tests/test_tasks.py

```diff
@@ -56,12 +59,26 @@ def set_up_course(self, enable_subsection_grades=True):
             ('course_id', unicode(self.course.id)),
             ('usage_id', unicode(self.problem.location)),
             ('only_if_higher', None),
+            ('modified_time', datetime.now().strftime("%y/%m/%d/%H/%M/%S/%f"))
         ])
 
         # this call caches the anonymous id on the user object, saving 4 queries in all happy path tests
         _ = anonymous_id_for_user(self.user, self.course.id)
         # pylint: enable=attribute-defined-outside-init,no-member
 
+    @contextmanager
+    def mock_student_module(self, timedelta_in_seconds=1):
+        """
+        Mocks the student module needed by the SCORE_PUBLISHED signal
+        handler. By default, sets the module's modified date in the past
+        (the happy path).
+        """
+        mock_student_module_return = MagicMock()
+        mock_student_module_return.modified = datetime.now() - timedelta(seconds=timedelta_in_seconds)
+        mock_student_module_return.modified = timezone.make_aware(mock_student_module_return.modified, timezone=None)
+        with patch("courseware.models.StudentModule.objects.get", return_value=mock_student_module_return):
+            yield
```

Comment:
Minor nit: would it make sense to have the default `timedelta_in_seconds` become `seconds_after_now`, with a default value of -1? That would allow tests that use an "in the future" value to just use 10 instead of the double-negation (-10 seconds *before* `now()`)